src/System.Text.Json/src/System/Text/Json/Node/JsonString.cs

```diff
@@ -67,6 +81,70 @@ public string Value
         /// <returns>The value represented by this instance.</returns>
         public override string ToString() => _value;
 
+        /// <summary>
+        ///   Converts the ISO 8601 text value of this instance to its <see cref="DateTime"/> equivalent.
+        /// </summary>
+        /// <returns>A <see cref="DateTime"/> equivalent to the text stored by this instance.</returns>
+        /// <exception cref="FormatException">
+        ///   Text value of this instance is not in an ISO 8601 defined DateTime format.
+        /// </exception>
+        public DateTime GetDateTime() => DateTime.ParseExact(_value, "s", CultureInfo.InvariantCulture);
+
+        /// <summary>
+        ///   Converts the ISO 8601 text value of this instance to <see cref="DateTimeOffset"/> equivalent.
+        /// </summary>
+        /// <returns>A <see cref="DateTimeOffset"/> equivalent to the text stored by this instance.</returns>
+        /// <exception cref="FormatException">
+        ///   Text value of this instance is not in an ISO 8601 defined DateTimeOffset format.
+        /// </exception>
+        public DateTimeOffset GetDateTimeOffset() => DateTimeOffset.ParseExact(_value, "s", CultureInfo.InvariantCulture);
+
+        /// <summary>
+        ///   Converts the text value of this instance to its <see cref="Guid"/> equivalent.
+        /// </summary>
+        /// <returns>A <see cref="Guid"/> equivalent to the text stored by this instance.</returns>
+        /// <exception cref="FormatException">
+        ///   Text value of this instance is not in a GUID recognized format.
+        /// </exception>
+        public Guid GetGuid() => Guid.Parse(_value);
```

Comment:
This supports all GUID formats, the rest of the JSON stack only supports the "D" format.  This should be restricted, too.